api-lib/gql/makeThunder.ts

```diff
@@ -1,15 +1,42 @@
 import { apiFetch, Thunder } from './__generated__/zeus';
 
-export const makeThunder = (url: string, adminSecret: string) => {
+export type UserOptions = {
+  profileId: number;
+  address: string;
+};
+
+type AdminOptions = {
+  adminSecret: string;
+};
+
+type Options = UserOptions | AdminOptions;
+
+type ThunderOptions = {
+  url: string;
+  options: Options;
+};
+
+const mapOptionsToHeaders = (options: UserOptions) => ({
+  'x-hasura-role': 'user',
+  'x-hasura-address': options.address,
+  'x-hasura-user-id': String(options.profileId),
```

Comment:
nit: use the .toString() method

Reply:
Wondering why is that preferable, refactored with https://github.com/coordinape/coordinape/pull/1486/commits/269330308f6017784c34ea4d8a369e31169cf3c5

Reply:
@karelianpie it's a matter of reliability in patterns in JS. Almost everything in JS has a `.toString` method, including Strings, so it's a reliable primitive for conversion. Using Class constructors for the purpose of conversion can have some weird side effects on the prototype chain, and can short circuit prototype inheritance, causing some weird bugs if you want to disambiguate something or do a comparison using the JS `typeof` or `instanceof` operators. 

I think it's insignificant in this case (and thus why it's a nit) but I like to reinforce good patterns when I have a chance.

Reply:
Thanks for the explanation! For this particular case as it is a header value, I thought we would probably want some safety in case the value is `null` or `undefined` which would just throw with `.toString()` but with a class would just be converted to `"null"` or `"undefined"` -- in general your argument makes sense, and I agree with you that for this case it's probably not relevant.

Reply:
@karelianpie yes, there are other considerations if the variable is nullish. Fortunately, those are logical exceptions to to the rule the the `toString` method and they throw a TypeError and more importantly, fail typechecking